k-distribution/src/main/scripts/bin/kunparse

```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/env bash
+set -e
+set -o pipefail
+set -u
+
+export PATH="$(cd "$(dirname "$0")"; pwd):$PATH"
+
+# initialize flags
+ARGV=()
+dir=.
+filterSubst=true
+literal=false
+outputFile=-
+outputMode=pretty
+result=1
+verbose=false
+
+# setup temp files
+now=`date +"%Y-%m-%d-%H-%M-%S"`
+tempDir="$(mktemp -d .krun-${now}-XXXXXXXXXX)"
+tempFiles=("$tempDir")
+trap 'rm -rf ${tempFiles[*]}' INT TERM EXIT
+
+execute () {
+  (
+  if $verbose; then
+    set -x
+  fi
+  "$@"
+  )
+}
+
+KUNPARSE=$(basename "$0")
+
+fold_lines="fold -s"
+
+error () {
+  printf "[Error] Critical: $@\n" | $fold_lines 1>&2
+  exit $result
+}
+
+warning () {
+  printf "[Warning] Compiler: $@\n" | $fold_lines 1>&2
+}
+
```

Comment:
This isn't used anywhere and I'm not convinced it would make sense as a warning message for this script.

In general, I would say both `error()` and `warning()` should probably have `kunparse` (or `$0`) somewhere in their output, so that we know where the message is coming from.